contract/security/contract_security.xml

```diff
@@ -0,0 +1,31 @@
+<odoo noupdate="1">
+
+    <record id="rule_contract_contract_multi_company" model="ir.rule">
+        <field name="name">Contract contract multi-company</field>
+        <field name="model_id" ref="model_contract_contract"/>
+        <field name="global" eval="True"/>
+        <field name="domain_force">['|', ('company_id', '=', False), ('company_id', 'child_of', [user.company_id.id])]</field>
```

Comment:
Better to use the new domains for version 13:
[('company_id', 'in', company_ids)]